dvc/commands/data.py

```diff
@@ -244,10 +245,16 @@ def add_parser(subparsers, parent_parser):
         help="Show untracked files.",
     )
     data_status_parser.add_argument(
-        "--remote-refresh",
+        "--not-in-remote",
         action="store_true",
         default=False,
-        help="Refresh remote index.",
+        help="Show files not in remote.",
+    )
+    data_status_parser.add_argument(
+        "--no-remote-refresh",
```

Comment:
@efiop I think refreshing is a more reasonable default because it is probably more intuitive and expected unless users are thinking deeply about how it compares to git. However, if you have a reason why you strongly prefer to rely on the index by default, I'm open to changing this back as long as we also keep the `--not-in-remote` option.

Reply:
If it's an explicit flag, fetching latest changes make sense to me. I'm okay with showing a hint/summary to `dvc push` if needed by default.